### ISSUE TITLE
Staggered GridView in Flutter

### ISSUE THREAD

**Issue by gurleensethi** [flutter/flutter#40856]
## Use case
Why isn't there in-built support for a staggered grid view in Flutter?
This is a basic use case that is covered by almost any mobile framework.
For example, Android has a `StaggeredGridLayoutManager` for `RecyclerView`.
I don't want to rely on a 3rd party library for this type of UI constructs.

**Comment by escamoteur** [comment]
Could you elaborate what a StaggeredGridView would work?

**Comment by gurleensethi** [comment]
Here is an image of what a Staggered Grid might look like.

![staggered_grid](https://i.stack.imgur.com/363QI.png)

**Comment by goderbauer** [comment]
Looks like the community has already provided a package for this: https://pub.dev/packages/flutter_staggered_grid_view

**Comment by gurleensethi** [comment]
@goderbauer Yes I know, but as I mentioned, I don't want to rely on a 3rd party package for this functionality, as I believe this kind of functionality should be provided by the framework itself. 

`flutter_staggered_grid_view` provides a really nice solution, but I think Flutter framework should have at least the basic support for a staggered grid.

**Comment by escamoteur** [comment]
@gurleensethi I fear if you want this soon you will have to go with the package :-(  Btw Romain the author does good stuff

**Comment by monnguyen2512** [comment]
@escamoteur The performance of this package is very low :(

**Comment by khalidlionel** [comment]
Any solutions for slow performance ?

**Comment by zmtzawqlp** [comment]
> Any solutions for slow performance ?

try https://pub.dev/packages/waterfall_flow

**Comment by khalidlionel** [comment]
@zmtzawqlp 
Thank you so much bro !
This solution worked for me .

**Comment by SirusCodes** [comment]
This could be closed @Hixie #59582

**Comment by 337240552** [comment]
> Looks like the community has already provided a package for this: https://pub.dev/packages/flutter_staggered_grid_view

serious performance issue when list is large！

**Comment by icnahom** [comment]
Unfortunatly, `flutter_staggered_grid_view`  have bad performance and `waterfall_flow` can't have varying width. So for now we will adopt our UI Design to SliverGrid's and hopefully wait for something solid from Flutter.

**Comment by letsar** [comment]
[flutter_staggered_grid_view](https://pub.dev/packages/flutter_staggered_grid_view) has been completely rewritten, and the performance of Masonry lists have been greatly improved.

**Comment by FrankyLee-dev** [comment]
Staggered GridView is very commonly used and effective in modern apps, whether ios' UICollectionView' or android's 'Recycleview' is very simple and convenient to implement. This sounds frustrating to introduce a third-party library to achieve this layout. Flutter is extremely powerful, I'm a flutter novice, I don't know how difficult it is to implement, or how difficult it is, but it's really needed. Hope to get a reply. Thanks. good luck.

**Comment by percula** [comment]
I would love to see this added to the Flutter SDK. I've tried the `flutter_staggered_grid_view` package and on the surface it seems to work great. But after using it, I've found it has many bugs including [this one](https://github.com/letsar/flutter_staggered_grid_view/issues/265) that is preventing me from using the package. 

Either this or #97544 would solve so many problems.
